crates/storage/src/schema/revision_0052.rs

```diff
@@ -0,0 +1,1748 @@
+            block_hash  BLOB NOT NULL,
+            tx          BLOB,
+            receipt     BLOB,
+            events      BLOB
+        )",
+        [],
+    )
+    .context("Creating starknet_transactions_new table")?;
+    let mut query_stmt =
+        tx.prepare("SELECT hash, idx, block_hash, tx, receipt FROM starknet_transactions")?;
+    let mut insert_stmt = tx.prepare(
+        r"INSERT INTO starknet_transactions_new (hash, idx, block_hash, tx, receipt, events)
+                                         VALUES (?, ?, ?, ?, ?, ?)",
+    )?;
+    const BATCH_SIZE: usize = 10_000;
+    let mut rows = query_stmt.query([])?;
+    let mut progress = 0;
+    loop {
+        let mut batch_size = 0;
+        for _ in 0..BATCH_SIZE {
+            match rows.next() {
+                Ok(Some(row)) => {
+                    let hash = row.get_ref_unwrap("hash").as_blob()?;
+                    let idx = row.get_ref_unwrap("idx").as_i64()?;
+                    let block_hash = row.get_ref_unwrap("block_hash").as_blob()?;
+                    let transaction = row.get_ref_unwrap("tx").as_blob()?;
+                    let receipt = row.get_ref_unwrap("receipt").as_blob()?;
+                    transform_tx
+                        .send((
+                            hash.to_vec(),
+                            idx,
+                            block_hash.to_vec(),
+                            transaction.to_vec(),
+                            receipt.to_vec(),
+                        ))
+                        .context("Sending transaction to transformer")?;
+                    batch_size += 1;
+                }
+                Ok(None) => break,
+                Err(err) => return Err(err.into()),
+            }
+        }
+        for _ in 0..batch_size {
+            if progress % 1000 == 0 && progress_logged.elapsed() > LOG_RATE {
+                progress_logged = Instant::now();
+                tracing::info!(
+                    "Migrating rows: {:.2}%",
+                    (progress as f64 / count as f64) * 100.0
+                );
+            }
+            let (hash, idx, block_hash, transaction, receipt, events) = insert_rx.recv()?;
+            insert_stmt.execute(params![hash, idx, block_hash, transaction, receipt, events])?;
+            progress += 1;
+        }
+        if batch_size < BATCH_SIZE {
+            // This was the last batch.
+            break;
+        }
+    }
+
+    drop(insert_tx);
+    drop(transform_tx);
+
+    // Ensure that all transformers have finished successfully.
+    for transformer in transformers {
+        transformer.join().unwrap();
+    }
+
+    tracing::info!("Dropping old starknet_transactions table");
+    tx.execute("DROP TABLE starknet_transactions", [])?;
+    tracing::info!("Renaming starknet_transactions_new to starknet_transactions");
+    tx.execute(
+        "ALTER TABLE starknet_transactions_new RENAME TO starknet_transactions",
+        [],
+    )?;
```

Comment:
Hmm, does this preserve the `starknet_transactions_block_hash` index we had for the old table? I think it doesn't.

Reply:
Yep, it's missing after the migration.

The whole migration took 28 minutes on my mainnet db. Re-creating the missing index (`CREATE INDEX starknet_transactions_block_hash ON starknet_transactions(block_hash)`) is then another 2.5 minutes (I've done that manually).

Reply:
Actually this reminds me. Do we want to change this block hash to a block number index instead? It would also be faster to lookup..